keps/sig-architecture/20191001-prevent-permabeta.md

```diff
@@ -0,0 +1,167 @@
+where users and vendors start building important infrastructure against APIs that are not considered stable.
+In addition, once a REST API is on by default, the incentive to further stabilize appears to diminish.
+See the REST API that have been beta for a long time: CSRs and Ingresses as examples.
+If we're honest with ourselves, a single actor has been cleaning up behind a lot of the project to unstick perma-beta APIs.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+
+### Goals
+
+1. Prevent k8s.io REST APIs from being in a single beta version for more than nine months.
+2. Prevent beta APIs from being treated as GA by users and vendors.
+
+### Non-Goals
+
+1. Promote APIs to GA before they are ready.
+2. Control non-k8s.io REST APIs.
+3. Control features that are not REST APIs.
+4. Control fields on otherwise GA REST APIs.
+
+## Proposal
+
+Once a REST API reaches beta, it has nine months to 
+1. reach GA and deprecate the beta or 
+2. have a new beta version and deprecate the previous beta.
+
+If neither of those conditions met, the beta REST API is deprecated in the second release with a stated intent to remove the REST API entirely.
+To avoid removal, the REST API must create a new beta version (it cannot go directly from deprecated to GA).
+
+For example, in v1.16, v1beta1 is released. Sample release note and API doc:
+> * "The v1beta1 version of this API will be evaluated during v1.16, v1.17, and v1.18, then deprecated in v1.19 (in favor of a new beta version, a GA version, or with no replacement), then removed in v1.22"
+
+Scenario A - progression to v1beta2 in v1.19. Sample release note and API doc:
+> * "The v1beta1 version of this API is deprecated in favor of v1beta2, and will be removed in v1.22"
+> * "The v1beta2 version of this API will be evaluated during v1.19, v1.20, and v1.21, then deprecated in v1.22 (in favor of a new beta version, a GA version, or with no replacement), then removed in v1.25"
+
+Scenario B - progression to v1 in v1.19. Sample release note and API doc:
+> * "The v1beta1 version of this API is deprecated in favor of v1, and will be removed in v1.22"
+
+Scenario C - deprecation with no replacement. Sample release note and API doc:
+> * "The v1beta1 version of this API is deprecated with no replacement, and will be removed in v1.22"
+
+By regularly having new beta versions, we can ensure that consumers will not grow long running dependencies on particular betas which could pin design decisions.
+It will also create an incentive for REST API authors to push their APIs to GA instead of letting them live in a permanent beta state.
+
+## Impacted APIs
+These sigs will need to announce in 1.19 that these APIs will be deprecated no later than 1.22 and removed no later than 1.25.
+This is the same as the standard for new beta APIs introduced in 1.19.
+
+### sig-apimachinery
+1. events.v1beta1.events.k8s.io
+
+### sig-apps
+1. jobtemplates.v1beta1.batch
```

Comment:
nit: jobtemplates isn't a standalone resource, it's a struct used within cronjob